kaiax/supply/impl/checkpoint.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2024 The Kaia Authors
+// This file is part of the Kaia library.
+//
+// The Kaia library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Kaia library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Kaia library. If not, see <http://www.gnu.org/licenses/>.
+
+package supply
+
+import (
+	"math/big"
+	"sync/atomic"
+	"time"
+
+	"github.com/kaiachain/kaia/kaiax/supply"
+)
+
+var (
+	checkpointInterval    = uint64(128)
```

Comment:
This gives a false impression that checkpoint is updated every 128 blocks; i.e., `getCheckpoint(128)=...=getCheckpoint(255)`.

Reply:
Sure. Will `checkpointCommitInterval` clarify it?

Reply:
That's better. btw, why did we name it checkpoint in the first place?
Maybe we're just saving TotalSupply in DB but with most fields removed just to reduce storage size.

Reply:
I think we were coming up with a different name to force re-accumulation after fixing a rewind-related bug. But AccReward sounds more accurate for the in-memory structure that increments every block. PTAL 8e14a9a.

Reply:
Modified to ReadAccReward(), WriteAccReward(), lastAccReward, etc.